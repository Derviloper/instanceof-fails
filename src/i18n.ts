import rawDe from "@/locales/de.json?raw";
import { createI18n } from "vue-i18n";
import { nextTick } from "vue";

const de = JSON.parse(rawDe);

export const SUPPORTED_LOCALES = [
  "de",
  "de-AT",
  "es",
  "fr",
  "fr-LU",
  "it",
  "nl",
  "nl-BE",
  "pl",
];

const i18n = createI18n({
  legacy: false,
  locale: "de",
  datetimeFormats: { de: de.datetimeFormats },
  messages: { de: de.messages },
  numberFormats: { de: de.numberFormats },
});

export default i18n;

export async function loadLocale(locale: string) {
  const config = await import(
    /* webpackChunkName: "locale-[request]" */ `./locales/${locale}.json`
  );
  i18n.global.setDateTimeFormat(locale, config.default.datetimeFormats);
  i18n.global.setLocaleMessage(locale, config.default.messages);
  i18n.global.setNumberFormat(locale, config.default.numberFormats);
  return nextTick();
}

export async function setI18nLanguage(locale: string): Promise<boolean> {
  if (!SUPPORTED_LOCALES.includes(locale)) {
    return false;
  }
  if (!i18n.global.availableLocales.includes(locale)) {
    await loadLocale(locale);
  }
  i18n.global.locale.value = locale;
  /**
   * NOTE:
   * If you need to specify the language setting for headers, such as the `fetch` API, set it here.
   * The following is an example for axios.
   *
   * axios.defaults.headers.common['Accept-Language'] = locale
   */
  document.querySelector("html")?.setAttribute("lang", locale);
  return true;
}
