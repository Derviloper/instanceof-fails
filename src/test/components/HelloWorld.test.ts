import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import HelloWorld from "@/components/HelloWorld.vue";
import i18n from "@/i18n";
import router from "@/router";
import FloatingVue from "floating-vue";

describe("HelloWorld.vue", () => {
  it.each([{ msg: "" }, { msg: "Hello World" }])(
    "should render with props %#",
    (props) => {
      const wrapper = mount(HelloWorld, {
        global: { plugins: [i18n, router, FloatingVue] },
        props,
      });
      console.log(wrapper.find("h3"));
      expect(wrapper.html()).toMatchSnapshot();
    }
  );
});
