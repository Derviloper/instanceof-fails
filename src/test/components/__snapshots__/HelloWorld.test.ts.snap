// Vitest Snapshot v1

exports[`HelloWorld.vue > should render with props 0 1`] = `
"<div class=\\"greetings\\" data-v-469af010=\\"\\">
  <h1 class=\\"green\\" data-v-469af010=\\"\\"></h1>
  <h3 data-v-469af010=\\"\\"> You’ve successfully created a project with <a target=\\"_blank\\" href=\\"https://vitejs.dev/\\" data-v-469af010=\\"\\">Vite</a> + <a target=\\"_blank\\" href=\\"https://vuejs.org/\\" data-v-469af010=\\"\\">Vue 3</a>. What's next? </h3>
</div>"
`;

exports[`HelloWorld.vue > should render with props 1 1`] = `
"<div class=\\"greetings\\" data-v-469af010=\\"\\">
  <h1 class=\\"green\\" data-v-469af010=\\"\\">Hello World</h1>
  <h3 data-v-469af010=\\"\\"> You’ve successfully created a project with <a target=\\"_blank\\" href=\\"https://vitejs.dev/\\" data-v-469af010=\\"\\">Vite</a> + <a target=\\"_blank\\" href=\\"https://vuejs.org/\\" data-v-469af010=\\"\\">Vue 3</a>. What's next? </h3>
</div>"
`;
